File: backend/app/schemas/hotlist.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class HotlistItemOut(BaseModel):
    id: int
    source: str
    rank: int
    title: str
    url: str
    description: Optional[str] = None
    language: Optional[str] = None
    stars_today: Optional[int] = None
    stars_total: Optional[int] = None
    forks: Optional[int] = None
    hot_date: date
    created_at: datetime

    class Config:
        from_attributes = True


class HotlistResponse(BaseModel):
    date: date
    source: str
    items: list[HotlistItemOut]


class FetchResponse(BaseModel):
    message: str
    count: int
    date: date

File: frontend/src/api/hotlist.ts
import api from './index'

export interface HotlistItem {
  id: number
  source: string
  rank: number
  title: string
  url: string
  description?: string
  language?: string
  stars_today?: number
  stars_total?: number
  forks?: number
  hot_date: string
  created_at: string
}

export interface HotlistResponse {
  date: string
  source: string
  items: HotlistItem[]
}

export const hotlistApi = {
  list(params: { source?: string; hot_date?: string }) {
    return api.get<any, HotlistResponse>('/api/hotlist/', { params })
  },
  fetch(source: string = 'github') {
    return api.post<any, { message: string; count: number; date: string }>(`/api/hotlist/fetch/`, null, { params: { source } })
  },
  sources() {
    return api.get<any, { sources: { key: string; name: string; url: string }[] }>('/api/hotlist/sources/')
  }
}
